### DISCUSSION TITLE
Can I exclude the cask installed through the rb file when issuing the brew upgrade command?

### DISCUSSION THREAD

**Discussion by circumeo** [Homebrew/discussions#4627] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.28-5-g9f9ae1e
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9f9ae1edeb07d0e7790f18fd540ca76228a3eec2
Last commit: 9 hours ago
Core tap JSON: 07 Jul 01:48 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4.1-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

<img width="1416" alt="스크린샷 2023-07-07 오전 10 55 40" src="https://github.com/Homebrew/discussions/assets/76243664/4530332b-b0b0-4922-97ac-26089b731fcd">

I understand that cask does not upgrade when using the brew upgrade command.
However, the older version of the calibre installed using the rb file is recognized as brew and is being upgraded together when using the brew upgrade command.
Is there any way to solve this??

### Relevant casks

calibre

**Comment by p-linnane** [comment]
I'm not quite sure what you're asking. You should be able to upgrade it by running `brew upgrade --cask calibre`. What happens when you run that?

**Reply by circumeo** [reply]
I am sorry that my English is not good enough to accurately convey the situation 😭
What I want is that when I run the **brew upgrade** command, the **calibre** doesn't become an upgrade.

**Reply by p-linnane** [reply]
That's not supported. It would be better to install calibre directly from upstream instead of using `brew` so you can remain on the version you want.

**Reply by circumeo** [reply]
Yeah, I understood what you said
But I know that the brew upgrade command, not the brew upgrade --cask command, does not let cask upgrade.
Then the calibre is cask, do you know why it is upgraded to the brew upgrade command?

**Reply by p-linnane** [reply]
From [the manual](https://docs.brew.sh/Manpage#upgrade-options-outdated_formulaoutdated_cask-), running `brew upgrade` will:
>Upgrade outdated casks and outdated, unpinned formulae using the same options they were originally installed with

If you just run `brew upgrade` it will attempt to upgrade everything that's installed and outdated.

**Reply by circumeo** [reply]
Thank you so much for letting me know!
It was really a big help 🙏
